src/ImageSharp.Web/ImageSharpTagHelper.cs

```diff
@@ -0,0 +1,363 @@
+        {
+            if (this.Width.HasValue)
+            {
+                commands.Add(ResizeWebProcessor.Width, this.Width.Value.ToString(this.parserCulture));
+                output.Attributes.SetAttribute(WidthAttributeName, this.Width);
+            }
+
+            if (this.Height.HasValue)
+            {
+                commands.Add(ResizeWebProcessor.Height, this.Height.Value.ToString(this.parserCulture));
+                output.Attributes.SetAttribute(HeightAttributeName, this.Height);
+            }
+
+            if (this.ResizeMode.HasValue)
+            {
+                commands.Add(ResizeWebProcessor.Mode, this.ResizeMode.Value.ToString());
+            }
+
+            if (this.Position.HasValue)
+            {
+                commands.Add(ResizeWebProcessor.Anchor, this.Position.Value.ToString());
+            }
+
+            if (this.Center.HasValue)
+            {
+                string xy = $"{this.Center.Value.Y.ToString(this.parserCulture)}{this.separator}{this.Center.Value.Y.ToString(this.parserCulture)}";
+                commands.Add(ResizeWebProcessor.Xy, xy);
+            }
+
+            if (this.PadColor.HasValue)
+            {
+                commands.Add(ResizeWebProcessor.Color, this.PadColor.Value.ToHex());
+            }
+
+            if (this.Compand.HasValue)
+            {
+                commands.Add(ResizeWebProcessor.Compand, this.Compand.Value.ToString(this.parserCulture));
+            }
+
+            if (this.Orient.HasValue)
+            {
+                commands.Add(ResizeWebProcessor.Orient, this.Orient.Value.ToString(this.parserCulture));
+            }
+
+            if (this.Sampler.HasValue)
+            {
+                commands.Add(ResizeWebProcessor.Sampler, this.Sampler.Value.ToString());
+            }
+        }
+
+        private void AddAutoOrientCommands(CommandCollection commands)
+        {
+            if (this.AutoOrient.HasValue)
+            {
+                commands.Add(AutoOrientWebProcessor.AutoOrient, this.AutoOrient.Value.ToString());
+            }
+        }
+
+        private void AddFormatCommands(CommandCollection commands)
+        {
+            if (this.Format.HasValue)
+            {
+                commands.Add(FormatWebProcessor.Format, this.Format.Value.ToString());
+            }
+        }
+
+        private void AddBackgroundColorCommands(CommandCollection commands)
+        {
+            if (this.BackgroundColor.HasValue)
+            {
+                commands.Add(BackgroundColorWebProcessor.Color, this.BackgroundColor.Value.ToHex());
+            }
+        }
+
+        private void AddQualityCommands(CommandCollection commands)
+        {
+            if (this.Quality.HasValue)
+            {
+                commands.Add(QualityWebProcessor.Quality, this.Quality.Value.ToString(this.parserCulture));
+            }
+        }
+
+        private static string AddQueryString(
```

Comment:
Can't you use [`QueryHelpers.AddQueryString()`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.webutilities.queryhelpers.addquerystring?view=aspnetcore-6.0) instead?

Reply:
Ah no, cos that expects a name value collection. This is copied straight from that anyway.